src/main/java/io/github/kawamuray/wasmtime/ImportType.java

```diff
@@ -0,0 +1,56 @@
+package io.github.kawamuray.wasmtime;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.experimental.Accessors;
+
+@Accessors(fluent = true)
+@AllArgsConstructor(access = AccessLevel.PACKAGE)
+public class ImportType {
+    public enum Type {
+        FUNC,
+        GLOBAL,
+        TABLE,
+        MEMORY,
+    }
+
+    @Getter
+    private final Type type;
+
+    @Getter(AccessLevel.PACKAGE)
+    private final Object typeObj;
+
+    @Getter
+    private final String module;
+
+    @Getter
+    private final String name;
+
+    private void ensureType(ImportType.Type expected) {
+        if (type != expected) {
+            throw new RuntimeException(
+                String.format("ImportType expected to have type %s but is actually %s", expected, type));
+        }
+    }
+
+    public FuncType func() {
+        ensureType(ImportType.Type.FUNC);
+        return (FuncType) typeObj;
+    }
+
+    public Global global() {
```

Comment:
So this `global`, `memory`, and `table` also needs to return its specific type descriptor, which doesn't exists yet right?
If you want to leave these types (their type description) support for future effort, let's just remove these methods for now to avoid confusing users.